proposals/10-egress-gateways.md

```diff
@@ -203,6 +199,204 @@ Those topics are covered in the separate  **[Payload Processing proposal](../7-p
+type BackendDestination struct {
+  // +required
+  Type BackendType `json:"type"`
+  // +optional
+  Ports []BackendPort `json:"ports,omitempty"`
+  // +optional
+  FQDN *FQDNBackend `json:"fqdn,omitempty"`
+  // Service *ServiceBackend `json:"service,omitempty"`
+  // IP *IPBackend `json:"ip,omitempty"`
+}
+
+// BackendType defines the type of the Backend destination.
+// +kubebuilder:validation:Enum=Fqdn;Ip;KubernetesService
+type BackendType string
+
+const (
+  // Fqdn represents a fully qualified domain name.
+  BackendTypeFqdn BackendType = "Fqdn"
+  // Ip represents an IP address.
+  BackendTypeIp BackendType = "Ip"
+  // KubernetesService represents a Kubernetes Service.
+  BackendTypeKubernetesService BackendType = "KubernetesService"
+)
+
+type BackendPort struct {
+  // Number defines the port number of the backend.
+  // +required
+  // +kubebuilder:validation:Minimum=1
+  // +kubebuilder:validation:Maximum=65535
+  Number uint32 `json:"number"`
+  // Protocol defines the protocol of the backend.
+  // +required
+  // +kubebuilder:validation:MaxLength=256
+  Protocol BackendProtocol `json:"protocol"`
+  // TLS defines the TLS configuration for the backend.
+  // +optional
+  TLS *BackendTLS `json:"tls,omitempty"`
+  // +optional
+  ProtocolOptions *BackendProtocolOptions `json:"protocolOptions,omitempty"`
+}
+
+// BackendProtocol defines the protocol for backend communication.
+// +kubebuilder:validation:Enum=HTTP;HTTP2;TCP;MCP
+type BackendProtocol string
+
+const (
+  BackendProtocolHTTP  BackendProtocol = "HTTP"
+  BackendProtocolHTTP2 BackendProtocol = "HTTP2"
+  BackendProtocolTCP   BackendProtocol = "TCP"
+  BackendProtocolMCP   BackendProtocol = "MCP"
+)
+
+type BackendTLS struct {
+  // Mode defines the TLS mode for the backend.
+  // +required
+  Mode BackendTLSMode `json:"mode"`
+  // SNI defines the server name indication to present to the upstream backend.
+  // +optional
+  SNI string `json:"sni,omitempty"`
+  // CaBundleRef defines the reference to the CA bundle for validating the backend's
+  // certificate.
+  // Defaults to system CAs if not specified.
+  // +optional
+  CaBundleRef []ObjectReference `json:"caBundleRef,omitempty"`
+
+  InsecureSkipVerify *bool `json:"insecureSkipVerify,omitempty"`
+
+  // ClientCertificateRef defines the reference to the client certificate for mutual
+  // TLS. Only used if mode is MUTUAL.
+  // +optional
+  ClientCertificateRef *SecretObjectReference `json:"clientCertificateRef,omitempty"`
+
+  SubjectAltNames []string `json:"subjectAltNames,omitempty"`
+}
+
+// BackendTLSMode defines the TLS mode for backend connections.
+// +kubebuilder:validation:Enum=Simple;Mutual;Passthrough;PlatformProvided;InsecureDisable
+type BackendTLSMode string
+
+const (
+  // Enable TLS with simple server certificate verification.
+  BackendTLSModeSimple BackendTLSMode = "Simple"
+  // Enable mutual TLS.
+  BackendTLSModeMutual BackendTLSMode = "Mutual"
```

Comment:
Not seeing BackendTLS support setting client key/certs

Reply:
I'm not entirely clear on where client config for mTLS should live, but expect that should be related to discussions in https://github.com/kubernetes-sigs/gateway-api/issues/4192 and/or https://github.com/kubernetes-sigs/gateway-api/pull/3876

/cc @howardjohn